cmd/quadlet/main.go

```diff
@@ -0,0 +1,246 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/quadlet/pkg/quadlet"
+	"github.com/containers/quadlet/pkg/systemdparser"
+)
+
+var verboseFlag bool
+var versionFlag bool
+var isUser bool
+
+var noKmsg bool = false
+var kmsgFile *os.File = nil
+
+func logToKmsg(s string) bool {
+	if noKmsg {
+		return false
+	}
+
+	if kmsgFile == nil {
+		f, err := os.OpenFile("/dev/kmsg", os.O_WRONLY, 0644)
+		if err != nil {
+			noKmsg = true
+			return false
+		}
+		kmsgFile = f
+	}
+
+	if _, err := kmsgFile.Write([]byte(s)); err != nil {
+		kmsgFile.Close()
+		kmsgFile = nil
+		return false
+	}
+
+	return true
+}
+
+func Log(format string, a ...any) {
+	s := fmt.Sprintf(format, a...)
+	line := fmt.Sprintf("quadlet-generator[%d]: %s", os.Getpid(), s)
+
+	if !logToKmsg(line) {
+		// If we can't log, print to stderr
+		fmt.Fprintf(os.Stderr, "%s\n", line)
+		os.Stderr.Sync()
+	}
+}
+
+var debugEnabled bool = false
+
+func enableDebug() {
+	debugEnabled = true
+}
+
+func Debug(format string, a ...any) {
+	if debugEnabled {
+		Log(format, a...)
+	}
+}
+
+func getUnitDirs(user bool) []string {
```

Comment:
I'd appreciate a verbose comment on this function (and many others) to be able to match intent vs. implementation.

Other container tools such as Podman et al. will read system configurations (/etc/containers/) even when being run as a ordinary rootless user.  `man containers.conf`, for example, will read `/etc/containers/containers.conf` unless there is a `$HOME/.config/containers/containers.conf`.

I do not have an opinion on how Quadlet should handle that but I think should be documented in the code.